### PR TITLE
Gronsfeld Cipher

### PR DESCRIPTION
**Pull request by dennistreysa**
I added the Gronsfeld Cipher which is basically Vigenere with digits (0..9 = A..J).
Also a function to convert a string with digits to a string containing the alpha equivalents was added.

### CODE REVIEW

File: lib/utils/common_utils.dart [hunk 1]
```diff
@@ -25,6 +25,31 @@ String intListToString(List<int> list, {String delimiter: ''}) {
   return list.map((elem) => elem == null ? unknownElement : elem).join(delimiter).trim();
 }
 
+String digitsToAlpha(String text, {int aValue: 0, bool removeNonDigits: true}) {
```
Comment (S-Man42) [inline lib/utils/common_utils.dart]: I would add a test for this (and maybe for all missing util methods as well :D)

File: lib/widgets/tools/crypto/gronsfeld.dart [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+      children: <Widget>[
+        GCWTextField(
+          controller: _inputController,
+          onChanged: (text) {
+            setState(() {
+              _currentInput = text;
+            });
+          },
+        ),
+        GCWTextField(
```
Comment (S-Man42) [inline lib/widgets/tools/crypto/gronsfeld.dart]: You could use GCWIntegerTextField, couldn't you? And then: _currentKey = value['text']

File: lib/utils/common_utils.dart (hunk 1 above)
Comment (dennistreysa) [reply lib/utils/common_utils.dart]: I added a test.

File: lib/widgets/tools/crypto/gronsfeld.dart (hunk 2 above)
Comment (dennistreysa) [reply lib/widgets/tools/crypto/gronsfeld.dart]: Sure. I just hadn't figured out that there is a "base" folder and all the derived classed are outside of that. I changed it to GCWIntegerTextField which is indeed more convenient.

**Pull request merged**